projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt

```diff
@@ -57,7 +59,68 @@ class CommunityApiController(
             log.error("Exception thrown when calling ${communityApiUrl + request.requestURI}. community-api returned ${ex.message}")
             ResponseEntity.status(ex.statusCode)
                 .headers(ex.responseHeaders)
-                .body(ex.responseBodyAsString);
+                .body(ex.responseBodyAsString)
+        }
+    }
+
+    @PostMapping("/compare")
+    fun compare(@RequestBody compare: Compare, request: HttpServletRequest): CompareReport {
+        val headers = request.headerNames.asSequence().associateWith { request.getHeader(it) }
+            .filter { it.key.lowercase() == "Authorization".lowercase() }.toMutableMap()
+
+        headers.forEach { log.info("Header name: ${it.key}") }
+
+        val uri = try {
+            Uri.valueOf(compare.uri)
+        } catch (ex: Exception) {
+            return compare.toReport("This endpoint is not configured")
+        }
+        val ccdJsonString = try {
+            compareService.toJsonString(compare)
+        } catch (ex: Exception) {
+            return compare.toReport("${uri.ccdFunction} bean cannot be found. Has this been implemented yet?")
+        }
+        val comApiUri = uri.comApiUrl.replace("{crn}", compare.crn)
+        val comApiJsonString = try {
+            communityApiClient.proxy(URI.create(communityApiUrl + comApiUri), headers).body!!
+        } catch (ex: HttpStatusCodeException) {
+            log.error("Exception thrown when calling ${communityApiUrl + comApiUri}. community-api returned ${ex.message}")
+            return compare.toReport(ex.message ?: "No message")
         }
+        val ccdJson = Json.createReader(StringReader(ccdJsonString)).readValue().asJsonObject()
+        val comApiJson = Json.createReader(StringReader(comApiJsonString)).readValue().asJsonObject()
+        val diff: JsonPatch = Json.createDiff(ccdJson, comApiJson)
```

Comment:
Did you consider using [JSONAssert](https://github.com/skyscreamer/JSONassert#error-messages), or any other JSON comparison library for this?  Happy to continue with the javax dependencies and jsonpatch - just wondering.

Reply:
I found that JsonPatch provided a flexible way of looking at the differences. So just went with that